src/main/java/com/szymon_kaluza/protobuf_avro/application/AvroLibraryService.java

```diff
@@ -15,14 +16,14 @@ public class AvroLibraryService {
 
     public byte[] serialize(Library library) {
         SpecificDatumWriter<Library> writer = new SpecificDatumWriter<>(Library.class);
-        byte[] data = new byte[0];
-        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        NonCopyingByteArrayOutputStream stream = new NonCopyingByteArrayOutputStream(1024);
```

Comment:
Prevents extra byte array creations and copies which are not the fault of Avro :)